examples/swaps/swap.ts

```diff
@@ -77,30 +90,50 @@ const swap = async () => {
     // Get up to date swap result by querying onchain
     const queryOutput = await swap.query(rpcUrl);
 
+    // const client = createPublicClient({
+    //     chain: polygon,
+    //     transport: http(),
+    // });
+
+    const buildCallInput = {
+        slippage,
+        deadline,
+        queryOutput,
+        sender,
+        recipient,
+        wethIsEth,
+    };
+
+    const client = createTestClient({
+        mode: 'anvil',
+        chain: CHAINS[chainId],
+        transport: http(rpcUrl),
+    })
+        .extend(publicActions)
+        .extend(walletActions);
+
+    const permit2 = await Permit2Helper.signSwapApproval({
+        ...buildCallInput,
+        client,
+        owner: sender,
+    });
```

Comment:
`createTestClient` with the extends is the only way to satisfy the `Permit2Helper` type requirements for `client`. Does that mean `Permit2Helper` is not intended for use by consumers?

Reply:
Also this fails with undefined address even if you pass in an `account` to the `createTestClient`

```
InvalidAddressError: Address "undefined" is invalid.

- Address must be a hex value of 20 bytes (40 hex characters).
- Address must match its checksum counterpart.

Version: viem@2.12.1
    at encodeAddress (/home/matthu/Desktop/Balancer/b-sdk/node_modules/.pnpm/viem@2.12.1_typescript@5.3.3/node_modules/viem/utils/abi/encodeAbiParameters.ts:215:32)
    at prepareParam (/home/matthu/Desktop/Balancer/b-sdk/node_modules/.pnpm/viem@2.12.1_typescript@5.3.3/node_modules/viem/utils/abi/encodeAbiParameters.ts:152:12)
    at prepareParams (/home/matthu/Desktop/Balancer/b-sdk/node_modules/.pnpm/viem@2.12.1_typescript@5.3.3/node_modules/viem/utils/abi/encodeAbiParameters.ts:124:25)
    at encodeAbiParameters (/home/matthu/Desktop/Balancer/b-sdk/node_modules/.pnpm/viem@2.12.1_typescript@5.3.3/node_modules/viem/utils/abi/encodeAbiParameters.ts:97:26)
    at encodeFunctionData (/home/matthu/Desktop/Balancer/b-sdk/node_modules/.pnpm/viem@2.12.1_typescript@5.3.3/node_modules/viem/utils/abi/encodeFunctionData.ts:93:28)
    at readContract (/home/matthu/Desktop/Balancer/b-sdk/node_modules/.pnpm/viem@2.12.1_typescript@5.3.3/node_modules/viem/actions/public/readContract.ts:106:38)
    at Object.readContract (/home/matthu/Desktop/Balancer/b-sdk/node_modules/.pnpm/viem@2.12.1_typescript@5.3.3/node_modules/viem/clients/decorators/public.ts:1842:41)
    at getNonce (/home/matthu/Desktop/Balancer/b-sdk/src/entities/permit2Helper/index.ts:188:33)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at getDetails (/home/matthu/Desktop/Balancer/b-sdk/src/entities/permit2Helper/index.ts:167:18) {
  details: undefined,
  docsPath: undefined,
  metaMessages: [
    '- Address must be a hex value of 20 bytes (40 hex characters).',
    '- Address must match its checksum counterpart.'
  ],
  shortMessage: 'Address "undefined" is invalid.',
  version: 'viem@2.12.1'
}
```

Reply:
> `createTestClient` with the extends is the only way to satisfy the `Permit2Helper` type requirements for `client`. Does that mean `Permit2Helper` is not intended for use by consumers?

Looks like so far we've only used `Permit2Helper` in tests. I'm not sure if we should be exposing this for consumers - wdyt? If we do then I guess we'll need to do a refactor to support non test clients.

Reply:
> Also this fails with undefined address even if you pass in an `account` to the `createTestClient`
> 
> ```
> InvalidAddressError: Address "undefined" is invalid.
> 
> - Address must be a hex value of 20 bytes (40 hex characters).
> - Address must match its checksum counterpart.
> 
> Version: viem@2.12.1
>     at encodeAddress (/home/matthu/Desktop/Balancer/b-sdk/node_modules/.pnpm/viem@2.12.1_typescript@5.3.3/node_modules/viem/utils/abi/encodeAbiParameters.ts:215:32)
>     at prepareParam (/home/matthu/Desktop/Balancer/b-sdk/node_modules/.pnpm/viem@2.12.1_typescript@5.3.3/node_modules/viem/utils/abi/encodeAbiParameters.ts:152:12)
>     at prepareParams (/home/matthu/Desktop/Balancer/b-sdk/node_modules/.pnpm/viem@2.12.1_typescript@5.3.3/node_modules/viem/utils/abi/encodeAbiParameters.ts:124:25)
>     at encodeAbiParameters (/home/matthu/Desktop/Balancer/b-sdk/node_modules/.pnpm/viem@2.12.1_typescript@5.3.3/node_modules/viem/utils/abi/encodeAbiParameters.ts:97:26)
>     at encodeFunctionData (/home/matthu/Desktop/Balancer/b-sdk/node_modules/.pnpm/viem@2.12.1_typescript@5.3.3/node_modules/viem/utils/abi/encodeFunctionData.ts:93:28)
>     at readContract (/home/matthu/Desktop/Balancer/b-sdk/node_modules/.pnpm/viem@2.12.1_typescript@5.3.3/node_modules/viem/actions/public/readContract.ts:106:38)
>     at Object.readContract (/home/matthu/Desktop/Balancer/b-sdk/node_modules/.pnpm/viem@2.12.1_typescript@5.3.3/node_modules/viem/clients/decorators/public.ts:1842:41)
>     at getNonce (/home/matthu/Desktop/Balancer/b-sdk/src/entities/permit2Helper/index.ts:188:33)
>     at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
>     at getDetails (/home/matthu/Desktop/Balancer/b-sdk/src/entities/permit2Helper/index.ts:167:18) {
>   details: undefined,
>   docsPath: undefined,
>   metaMessages: [
>     '- Address must be a hex value of 20 bytes (40 hex characters).',
>     '- Address must match its checksum counterpart.'
>   ],
>   shortMessage: 'Address "undefined" is invalid.',
>   version: 'viem@2.12.1'
> }
> ```

You can track back to find the source of this, see if you can figure out and let me know if you can't.

Reply:
> Looks like so far we've only used Permit2Helper in tests. I'm not sure if we should be exposing this for consumers - wdyt? If we do then I guess we'll need to do a refactor to support non test clients.

Was originally excited for `Permit2Helper`, but on second thought the cost of refactor plus abstracting away the definition of which router is spender and when signature expires is probably not ideal

Reply:
Some more realizations:

1. Should we have seperate `swapV2` and `swapV3` examples since only v3 uses permit2?
2.  Permit2 examples will have to be on Sepolia cus that's the only chain Router is deployed
3. The token pair from scaffold prod I've been leaning on does not get picked up by SOR paths fetched from API